Tools/Scripts/webkitpy/layout_tests/models/test_configuration.py

```diff
@@ -31,15 +31,17 @@
 
 
 class TestConfiguration(object):
-    def __init__(self, version, architecture, build_type):
+    def __init__(self, version, architecture, build_type, build_style='normal'):
```

Comment:
I think the consequences of adding `build_style` are more dramatic than what's reflected in this change. For one, we'd need to pipe that same information through to the results database as a new configuration column.

What we've done in other places (namely, the results database) is treat `ASAN` and `Guard-Malloc` as `build_type`s. For `ASAN`, that is, strictly speaking, true. It's not exactly true for `Guard-Malloc` (ie, you can have `release`+`Guard-Malloc` or `debug`+`Guard-Malloc`, but in practice, we only ever run `Guard-Malloc` one way, and it seems unlikely that we would ever need to differentiate between `debug` and `release` Guard-Malloc results. I think that would turn this change into a much smaller one, we'd just need to add `asan` and `guard-malloc` to `ALL_BUILD_TYPES`, I believe. Is there a reason we aren't solving this problem that way?

Reply:
> I think the consequences of adding build_style are more dramatic than what's reflected in this change. For one, we'd need to pipe that same information through to the results database as a new configuration column.

That's already done in base.configuration_for_upload:
```python
    def configuration_for_upload(self, host=None):
        from webkitpy.results.upload import Upload

        configuration = self.test_configuration()
        host = self.host or host

        if self.get_option('guard_malloc'):
            style = 'guard-malloc'
        elif self._config.asan:
            style = 'asan'
        else:
            style = configuration.build_type

        return Upload.create_configuration(
            platform=host.platform.os_name,
            version=str(host.platform.os_version),
            version_name=host.platform.os_version_name(INTERNAL_TABLE) or host.platform.os_version_name(),
            architecture=configuration.architecture,
            style=style,
            sdk=host.platform.build_version(),
            flavor=self.get_option('result_report_flavor'),
            model=self.get_option('model'),
        )
```